### DISCUSSION TITLE
apptainer with neurodesk on super computer with no sudo

### DISCUSSION THREAD

**Discussion by seaguldee** [neurodesk/neurodesk.github.io-hugo#273] [Q&A] [answered]
Hi, 
I am tired of trying to run AFNI off of a container as it keeps crashing on our university supercomputer and am looking at just using neurodesk.  If I could get it to work, it would be heaven.  Unfortunately, our university just stop supporting singularity and now we are using apptainer (I think this is is just a new version of singularity).  I also DO NOT have sudo privileges and am far from an expert on running containers at all.  Sooo, that being said, when I try to just run neurodesk through apptainer, I type...

apptainer run docker://vnmd/neurodesktop:latest 

I get the following error???  Is this fixable for someone that is useless with containers and just honestly wants to be able to type "afni" in a terminal and run some code :(

sudo: /usr/bin/sudo must be owned by uid 0 and have the setuid bit set

**Comment by civier** [comment, marked as answer]
Dear @seaguldee

The Neurodesktop component of Neurodesk is a docker container, and thus, cannot be run on a supercomputer. This is why you receive the error message.

That said, you can install the neurocommand command-line interface (https://www.neurodesk.org/docs/neurodesktop/getting-started/hpc/) that will let you access Neruodesk's apptainer containers from the command line, including Neurodesk's AFNI container (You are right that apptainer is only a rebranding of Singularity). If that is what you're looking for, please try it and report back if you have any issues.

If your use case is different, please clarify.

Good luck!
Oren

**Comment by stebo85** [comment]
Dear @seaguldee,

To use Neurodesk on an HPC you have 2 options:

1) You can directly download the application containers to use in singularity/apptainer (apptainer is a fork of singularity and they are mostly compatible with each other). For this, you can follow the documentation here: https://www.neurodesk.org/docs/neurocontainers/singularity/

To get the latest afni container you can run: 
```
curl -X GET https://objectstorage.us-ashburn-1.oraclecloud.com/n/sd63xuke79z3/b/neurodesk/o/afni_23.0.07_20230302.simg -O
```

Then you can shell into the container:
apptainer shell afni_23.0.07_20230302.simg

2) The second option is to install neurocommand on your HPC (https://www.neurodesk.org/docs/neurodesktop/getting-started/hpc/) as mentioned by Oren in the reply above. For this you do not need sudo access - only apptainer.

Kind regards
Steffen
